types/mongoose-delete/index.d.ts

```diff
@@ -0,0 +1,125 @@
+// Type definitions for mongoose-delete 0.5
+// Project: https://github.com/dsanel/mongoose-delete
+// Definitions by: Mochamad Arifin <https://github.com/ndunks>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.2
+
+import mongoose = require('mongoose');
+
+declare global {
```

Comment:
why is this in the global namespace? The instructions for mongoose-delete always use commonjs require. They don't mention browser usage.

Reply:
Because I don't know how to export interfaces and a function (when the module just use `module.exports = function()`. But I need to exports defined interfaces too. Can you send me example or guide to solve this? I'm new in typing definition.

Reply:
You got it almost right. =) `namespace mongoose_delete` will merge with `function mongoose_delete` and `export = mongoose_delete` will export both. You just need to delete the enclosing `declare global`.

Reply:
I’ll try and test it.. thanks for your attention..